src/WebJobs.Script.WebHost/Middleware/HstsConfigurationMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.HttpsPolicy;
+using Microsoft.Azure.WebJobs.Script.WebHost.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Middleware
+{
+    public class HstsConfigurationMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IScriptWebHostEnvironment _webHostEnvironment;
+        private HstsMiddleware _hstsMiddleware;
+        private HostHstsOptions _hostHstsOptions;
+
+        public HstsConfigurationMiddleware(RequestDelegate next, IScriptWebHostEnvironment webHostEnvironment)
+        {
+            _next = next;
+            _webHostEnvironment = webHostEnvironment;
+        }
+
+        public async Task Invoke(HttpContext httpContext, IOptions<HostHstsOptions> options)
+        {
+            if (_hostHstsOptions != options.Value)
```

Comment:
If this value changes (like during specialization) during a wave of requests, you'll end up with a race here and we'll be creating a bunch of new `HstsMiddleware` objects. Need to come up with a locking strategy that lets us create this whenever the options have changed without instantiating a bunch of times.

Reply:
Yeah the race condition would be bad, May i can use the IOptionsMonitor instead of IOptions and only create an object on change. Will update this logic.

Reply:
Yes, this needs to be synchronized.